### DISCUSSION TITLE
Support 200.html for single page apps and client-side routing

### DISCUSSION THREAD

**Discussion by newinnovations** [encode/starlette#2116] [Ideas] [answered]
For single page apps and client-side routing several web servers support a 200.html page, being served when the path cannot be found with a status code of 200. This is similar to the 404.html already supported by staticfiles.py. The 404.html can be used for this purpose but it will raise errors in the console because of the 404 status code.

I made a simple addition to staticfiles.py and tested it in my own fork.

**Comment by Kludex** [comment]
What's wrong with `index.html`?

**Reply by newinnovations** [reply, marked as answer]
Nothing wrong with `index.html`, but it won't get served on non existing paths.

With client-side routing the URL could be at `/site/static/some/client/side/url` and when the user refreshes the page you want to serve the main page which loads the JavaScript application again. Then the application can handle the client side route from there.

Probably better explained here: https://surge.sh/help/adding-a-200-page-for-client-side-routing

**Reply by alex-oleshkevich** [reply]
I think your question is not about static files. It is about routing.
This is usually solvable with a catch-all route 
```python
Route('/static/{path:path}', my_200_html_handler)
```

**Reply by newinnovations** [reply]
That won't serve existing files under the /static path. And we do need to serve those. The 200.html is only for the client side routes that don't exist at the server side.
